src/graphql/queries.js

```diff
@@ -17,75 +17,155 @@
 
 // Code related to GraphQL queries, fragments, variables, etc.
 
+import gql from 'graphql-tag'
+// eslint-disable-next-line no-unused-vars
+import { DocumentNode } from 'graphql'
+
 // IMPORTANT: queries here may be used in the offline mode to create mock data. Before removing or renaming
 // queries here, please check under the services/mock folder for occurrences of the variable name.
 
 /**
- * Query used to retrieve data for a workflow Tree view.
- * @type {string}
+ * @type {DocumentNode}
  */
-export const WORKFLOW_TREE_QUERY = `
-subscription {
-  workflows(ids: ["WORKFLOW_ID"]) {
-    id
-    name
-    status
-    owner
-    host
-    port
-    cyclePoints: familyProxies(ids: ["root"]) {
+const WORKFLOW_TREE_DELTAS_SUBSCRIPTION = gql`
+subscription OnWorkflowDeltasData ($workflowId: ID) {
+  deltas (workflows: [$workflowId], stripNull: true) {
+   ...WorkflowTreeDeltas
+  }
+}
+
+# TREE DELTAS BEGIN
+
+fragment WorkflowTreeDeltas on Deltas {
+  id
+  shutdown
+  added {
+    ...WorkflowTreeAddedData
+  }
+  updated {
+    ...WorkflowTreeUpdatedData
+  }
+  pruned {
+    ...WorkflowTreePrunedData
+  }
+}
+
+fragment WorkflowTreeAddedData on Added {
+  workflow {
+    ...WorkflowData
+    cyclePoints: familyProxies(ids: ["root"], ghosts: true) {
       cyclePoint
     }
-    taskProxies(sort: { keys: ["cyclePoint"] }) {
-      id
-      name
-      state
-      isHeld
-      cyclePoint
-      latestMessage
-      firstParent {
-        id
-        name
-        cyclePoint
-        state
-      }
-      task {
-        meanElapsedTime
-        name
-      }
+    taskProxies(sort: { keys: ["cyclePoint"] }, ghosts: true) {
+      ...TaskProxyData
```

Comment:
I like these fragments 👏 

This is a really nice way to get flat subscriptions.

Getting ahead of myself and thinking about what the subscription interface might look like in the future:

* Components could provide lists of fields to the workflow service.
* This would give us total separation of components and GraphQL implementation.
* We can potentially do some clever things like:
  * Validating the requested fields against the GraphQL schema.
  * Providing sensible fallback values in the event that the schema doesn't contain the required fields.

Reply:
They really simplify queries, kudos to @dwsutherland who showed me fragments first (quite a while ago I think?).

>This would give us total separation of components and GraphQL implementation.

+1 !

>Components could provide lists of fields to the workflow service.

Like with `props` as it's doing now? Or something else?

I think it will be easier to discuss the improvements to the subscription interface with code, but +1 to any improvement over the current code. It is mostly a "let's get deltas working" approach. 

In the same sense, the Workflow view & component were a "let me see if lumino would work here..."... that just worked and is still in use 😄